agent/consul/rpc.go

```diff
@@ -370,10 +371,34 @@ func (s *Server) globalRPC(method string, args interface{},
 	return nil
 }
 
+type raftEncoder func(structs.MessageType, interface{}) ([]byte, error)
+
 // raftApply is used to encode a message, run it through raft, and return
 // the FSM response along with any errors
 func (s *Server) raftApply(t structs.MessageType, msg interface{}) (interface{}, error) {
```

Comment:
Is the intent here to lazily deprecate plain `raftApply`?

Reply:
Yes. Basically now that we have the option for protobuf or msgpack encoding I think it would be best to opt-in to one or the other. To keep the changes from being massive thought I changed `raftApply` to call `raftApplyMsgpack`.